src/redux/reducer.js

```diff
@@ -0,0 +1,41 @@
+import { fromJS } from 'immutable';
+import C from './constants';
+import initialState from './initial_state';
+
+export default (state = fromJS(initialState), action) => {
+  switch (action.type) {
+    case C.SELECT_ADD_SET: {
+      const { name } = action.data;
+      if (!name) return state;
+      return state.merge({
+        [name]: { isOpen: false, selected: [] },
+      });
+    }
+    case C.SELECT_TOGGLE_OPEN: {
+      const { name, open } = action.data;
+      return state.merge({
+        [name]: (state.get(name) || fromJS({})).merge({ isOpen: open }),
```

Comment:
not sure what `[name]` is here... and the `fromJS({})` – are we clearing the entire state and setting it to `action.data.open` if there is no `name`?